cezo_fl/shared.py

```diff
@@ -33,7 +33,7 @@ def update_model_given_seed_and_grad(
     for local_update_seed, local_update_grad_vector in zip(iteration_seeds, iteration_grad_scalar):
         # create gradient
         update_grad = get_update_grad_for_1_seed(
-            grad_estimator, local_update_grad_vector, local_update_seed
+            grad_estimator, local_update_grad_vector.to(grad_estimator.device), local_update_seed
```

Comment:
why do you need this `to`?

Reply:
Good catch, I did this hack to make things work fast. But removed this hack in the latest commit.

Now all cross device data transfer happens in 1 place: parallizable_client_job()